tests/e2e/reports/reports-sidebar-filter.wdio-spec.js

```diff
@@ -0,0 +1,70 @@
+const moment = require('moment');
+
+const utils = require('../../utils');
+const commonElements = require('../../page-objects/common/common.wdio.page');
+const reportsPage = require('../../page-objects/reports/reports.wdio.page');
+const loginPage = require('../../page-objects/login/login.wdio.page');
+const userFactory = require('../../factories/cht/users/users');
+const placeFactory = require('../../factories/cht/contacts/place');
+const personFactory = require('../../factories/cht/contacts/person');
+const reportFactory = require('../../factories/cht/reports/generic-report');
+
+describe('Reports Sidebar Filter', () => {
+  let savedReports;
+  const today = moment();
+  const parent = placeFactory.place().build({ _id: 'dist1', type: 'district_hospital' });
+  const user = userFactory.build({ username: 'john', roles: [ 'chw' ] });
+  const patient = personFactory.build({ parent: { _id: user.place._id, parent: { _id: parent._id } } });
+  const reports = [
+    reportFactory.build(
+      { form: 'P', reported_date: moment([today.year(), today.month() - 4, 1, 23, 30]).valueOf() },
+      { patient, submitter: user.contact, fields: { lmp_date: 'Feb 3, 2022' },
+      }),
+    reportFactory.build(
+      { form: 'P', reported_date: moment([today.year(), today.month() - 1, 16, 0, 30]).valueOf() },
+      { patient, submitter: user.contact, fields: { lmp_date: 'Feb 16, 2022' },
+      }),
+    reportFactory.build(
+      { form: 'V', reported_date: moment([today.year(), today.month() - 5, 15, 0, 30]).valueOf() },
+      { patient, submitter: user.contact, fields: { ok: 'Yes!' },
+      }),
+    reportFactory.build(
+      { form: 'V', reported_date: moment([today.year(), today.month() - 1, 18, 23, 30]).valueOf() },
+      { patient, submitter: user.contact, fields: { ok: 'Yes!' },
+      }),
+  ];
+
+  beforeEach(async () => {
+    const settings = await utils.getSettings();
+    const permissions = {
+      ...settings.permissions,
+      can_view_sidebar_filter: [ 'chw' ]
+    };
+    await utils.updateSettings({ permissions }, true);
+
+    const results = await utils.saveDocs([ parent, patient, ...reports ]);
+    results.splice(0, 2); // Keeping only reports
+    savedReports = results.map(result => result.id);
+
+    await utils.createUsers([ user ]);
+    await loginPage.login(user);
+    await commonElements.goToReports();
+  });
+
+  it('should filter by date', async () => {
+    await (await reportsPage.firstReport()).waitForDisplayed();
+    await reportsPage.openSidebarFilter();
+    await reportsPage.openSidebarFilterAccordion('#date-filter-accordion');
+
+    await reportsPage.setSidebarFilterDate('#fromDateFilter');
+    await reportsPage.setSidebarFilterDate('#toDateFilter', 2, 'r3c5');
```

Comment:
I think it will be better if we keep the "selectors" in the `*.page` file. Could you please create constants for those selectors in the `reports.wdio.page.js` file.